### ISSUE TITLE
should url decode the search index pattern

### ISSUE THREAD

**Issue by tuziben** [quickwit-oss/quickwit#4291]
**Describe the bug**
When we use the Elastic or OpenSearch Golang clients to search some indexes in this pattern: `gh-archive*`, will get a error message: `"message":"index ID pattern `gh-archive%2A` is invalid. patterns must match the following regular expression: `^[a-zA-Z\\*][a-zA-Z0-9-_\\.\\*]{0,254}$`"`

**Steps to reproduce (if applicable)**

1. quickwit built with this hash [8a40a4d143937aacdbc5206d57f3ef7ea26ed9bb](https://github.com/quickwit-oss/quickwit/commit/8a40a4d143937aacdbc5206d57f3ef7ea26ed9bb)
2. config in quickwit

```
rest:
  listen_port: 7280
  cors_allow_origins:
    - "*"
  extra_headers:
    x-elastic-product: Elasticsearch
```

use the Elastic and Opensearch client to query Quickwit
Golang code
```go
       // elastic client 7.x, 8.x
       // opensearch 2.*

 	client, err := elasticsearch.NewClient(elasticsearch.Config{
		Addresses: []string{"http://127.0.0.1:7280/api/v1/_elastic"},
	})

	if err != nil {
		fmt.Println(err)
	}

	index := "gh-archive*"
	queryBody := "{\n    \"query\": {\n        \"bool\": {\n            \"filter\": [\n                {\n                    \"match_phrase\": {\n                        \"id\": {\n                            \"query\": \"32230578641\"\n                        }\n                    }\n                }\n            ]\n        }\n    }\n}"
	res, err := client.Search(
		client.Search.WithIndex(index),
		client.Search.WithBody(strings.NewReader(queryBody)),
	)

	if err != nil {
		fmt.Println(err.Error())
		return
	}

	fmt.Println(res)

        {
          "message": "index ID pattern `gh-archive%2A` is invalid. patterns must match the following regular expression: `^[a-zA-Z\\*][a-zA-Z0-9-_\\.\\*]{0,254}$`"
        }
```

This issue seems unique to the Golang client; clients in other programming languages like Python and Java don't have this problem. I've looked through the Golang client's code and couldn't find any part where it's using URL encoding. But, based on the output we're seeing, it looks like the Golang client is URL encoding the asterisk (*).

So, to fix this, I think Quickwit should decode the URLs in all its search indices to get them back to their original format.
https://github.com/quickwit-oss/quickwit/blob/8a40a4d143937aacdbc5206d57f3ef7ea26ed9bb/quickwit/quickwit-config/src/lib.rs#L125

**Comment by fmassot** [comment]
@tuziben can you give the quickwit version?

**Comment by tuziben** [comment]
> 1. quickwit built with this hash [8a40a4d143937aacdbc5206d57f3ef7ea26ed9bb](https://github.com/quickwit-oss/quickwit/commit/8a40a4d143937aacdbc5206d57f3ef7ea26ed9bb)

The main branch or any other version can reproduce this issue

**Comment by fmassot** [comment]
Fixed by #4292
